src/calculation/CompositeReportBuilder.ts

```diff
@@ -0,0 +1,209 @@
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    };
+  }
+
+  public addPatientResults(results: ExecutionResult<T>) {
+    // https://build.fhir.org/ig/HL7/cqf-measures/composite-measures.html#all-or-nothing-scoring
+    if (this.compositeScoringType === 'all-or-nothing') {
+      let inIpp = false;
+      let inDenom = false;
+      let inNumer = true;
+
+      results.componentResults?.forEach(componentResult => {
+        const ippResult = componentResult.populationResults?.find(
+          pop => pop.populationType === PopulationType.IPP
+        )?.result;
+        if (ippResult === true) {
+          inIpp = true;
+        }
+
+        const denomResult = componentResult.populationResults?.find(
+          pop => pop.populationType === PopulationType.DENOM
+        )?.result;
+        if (denomResult === true) {
+          inDenom = true;
+        }
+
+        const numerResult = componentResult.populationResults?.find(
+          pop => pop.populationType === PopulationType.NUMER
+        )?.result;
+        if (numerResult === false) {
+          inNumer = false;
+        }
+      });
+
+      if (inIpp) {
+        if (inDenom) {
+          this.compositeFraction.denominator++;
+          if (inNumer) {
+            this.compositeFraction.numerator++;
+          }
+        }
+      }
+    } else if (this.compositeScoringType === 'opportunity') {
+      // https://build.fhir.org/ig/HL7/cqf-measures/composite-measures.html#opportunity-scoring
+
+      let inIpp = false;
+      let inDenom = false;
+
+      results.componentResults?.forEach(componentResult => {
+        const ippResult = componentResult.populationResults?.find(
+          pop => pop.populationType === PopulationType.IPP
+        )?.result;
+        if (ippResult === true) {
+          inIpp = true;
+        }
+
+        const denomResult = componentResult.populationResults?.find(
+          pop => pop.populationType === PopulationType.DENOM
+        )?.result;
+        if (denomResult === true) {
+          inDenom = true;
+        }
+
+        const numerResults = componentResult.populationResults
+          ?.filter(pop => pop.populationType === PopulationType.NUMER)
+          ?.map(r => r.result);
+
+        if (inIpp) {
+          if (inDenom) {
+            this.compositeFraction.denominator++;
+          }
+          numerResults?.forEach(result => {
+            if (result) {
+              this.compositeFraction.numerator++;
+            }
+          });
+        }
+      });
+    } else if (this.compositeScoringType === 'linear') {
+      // https://build.fhir.org/ig/HL7/cqf-measures/composite-measures.html#patient-level-linear-combination-scoring
+
+      // Always increment the denominator for linear scoring when processing a patient result
+      this.compositeFraction.denominator++;
+
+      const [patientNumerCount, patientDenomCount] = results.componentResults?.reduce(
```

Comment:
This is a very big nit pick so feel free to ignore, but I feel like we were maybe a little nervous about users being aware that the composite measure population list is a tuple of [denom, numer]. I know this is separate but maybe to reinforce that patten this could be [patientDenomCount, patientNumerCount]. Thoughts?

Reply:
Ideally users will never see this code! But I'll update it for consistency

Reply:
https://github.com/projecttacoma/fqm-execution/pull/235/commits/d3657a34a5367c199725a566d599df445f8d025d